processor/processor.go

```diff
@@ -179,19 +194,143 @@ func (proc *HandleT) Setup(backendConfig backendconfig.BackendConfig, gatewayDB
 	}
 }
 
+type StoreErrorOutputT struct {
+	Location string
+	Error    error
+}
+
+func (proc *HandleT) storeErorrsToObjectStorage(jobs []*jobsdb.JobT) StoreErrorOutputT {
+	localTmpDirName := "/rudder-proc-err-logs/"
+
+	uuid := uuid.NewV4()
+	logger.Debug("[Processor: storeErorrsToObjectStorage]: Starting logging to object storage")
+
+	tmpDirPath, err := misc.CreateTMPDIR()
+	if err != nil {
+		panic(err)
+	}
+	path := fmt.Sprintf("%v%v.json", tmpDirPath+localTmpDirName, fmt.Sprintf("%v.%v.%v.%v", time.Now().Unix(), config.GetEnv("INSTANCE_ID", "1"), fmt.Sprintf("%v-%v", jobs[0].JobID, jobs[len(jobs)-1].JobID), uuid))
+
+	gzipFilePath := fmt.Sprintf(`%v.gz`, path)
+	err = os.MkdirAll(filepath.Dir(gzipFilePath), os.ModePerm)
+	if err != nil {
+		panic(err)
+	}
+	gzWriter, err := misc.CreateGZ(gzipFilePath)
+	if err != nil {
+		panic(err)
+	}
+	defer os.Remove(gzipFilePath)
+
+	var contentSlice [][]byte
+	for _, job := range jobs {
+		rawJob, err := json.Marshal(job)
+		if err != nil {
+			panic(err)
+		}
+		contentSlice = append(contentSlice, rawJob)
+	}
+	content := bytes.Join(contentSlice[:], []byte("\n"))
+	gzWriter.Write(content)
+	gzWriter.CloseGZ()
+
+	outputFile, err := os.Open(gzipFilePath)
+	if err != nil {
+		panic(err)
+	}
+	prefixes := []string{"rudder-proc-err-logs", time.Now().Format("01-02-2006")}
+	uploadOutput, err := proc.errFileUploader.Upload(outputFile, prefixes...)
+
+	return StoreErrorOutputT{
+		Location: uploadOutput.Location,
+		Error:    err,
+	}
+}
+
+func (proc *HandleT) setErrJobStatus(jobs []*jobsdb.JobT, output StoreErrorOutputT) {
+	var statusList []*jobsdb.JobStatusT
+	for _, job := range jobs {
+		state := jobsdb.Succeeded.State
+		errorResp := []byte(`{"success":"OK"}`)
+		if output.Error != nil {
+			var err error
+			errorResp, err = json.Marshal(struct{ Error string }{output.Error.Error()})
+			if err != nil {
+				panic(err)
+			}
+			if job.LastJobStatus.AttemptNum >= maxFailedCountForErrJob {
+				state = jobsdb.Aborted.State
+			} else {
+				state = jobsdb.Failed.State
+			}
+		}
+		status := jobsdb.JobStatusT{
+			JobID:         job.JobID,
+			AttemptNum:    job.LastJobStatus.AttemptNum + 1,
+			JobState:      state,
+			ExecTime:      time.Now(),
+			RetryTime:     time.Now(),
+			ErrorCode:     "",
+			ErrorResponse: errorResp,
+		}
+		statusList = append(statusList, &status)
+	}
+	proc.errorDB.UpdateJobStatus(statusList, nil, nil)
```

Comment:
How about just dumping to errorDB and let the errored jobs be backed up from jobsdb?